pkg/gw/safeset_handler_test.go

```diff
@@ -0,0 +1,125 @@
+package gw
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	immuclient "github.com/codenotary/immudb/pkg/client"
+	immudb "github.com/codenotary/immudb/pkg/server"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+)
+
+func generateRandomTCPPort() int {
+	rand.Seed(time.Now().UnixNano())
+	min := 1024
+	max := 64000
+	return rand.Intn(max - min + 1)
+}
+
+func TestSafeset(t *testing.T) {
+	key := base64.StdEncoding.EncodeToString([]byte("Pablo"))
+	value := base64.StdEncoding.EncodeToString([]byte("Picasso"))
+	tcpPort := generateRandomTCPPort()
+	op := immudb.DefaultOptions().WithPort(tcpPort).WithDir("db_" + strconv.FormatInt(int64(tcpPort), 10))
+	s := immudb.DefaultServer().WithOptions(op)
+	go s.Start()
+	time.Sleep(2 * time.Second)
+	defer s.Stop()
```

Comment:
When I launch test on my machine It return following error:
```
 WARNING: LOG Compact FAILED with error: open db_57655/immudb/000002.sst: no such file or directory
While opening new table: 2
```
I belive that folder is deleted before badger stop:
try with something:
```
defer func() {
		s.Stop()
		os.RemoveAll(op.Dir)
	}()
```

Reply:
Yes, already fixed for #264 in 
https://github.com/codenotary/immudb/pull/274/commits/7e4d9789485a4c43755dd3a72450580e1de85f71
Will do the same here